Sources/MuxUploadSDK/PublicAPI/UploadManager.swift

```diff
@@ -100,27 +98,33 @@ public final class UploadManager {
     }
     
     internal func acknowledgeUpload(id: String) {
-        if let uploader = uploadersByID[id] {
+        if let uploader = uploadsByID[id] {
+            uploadsByID.removeValue(forKey: id)
             uploader.cancel()
         }
-        uploadersByID.removeValue(forKey: id)
         Task.detached {
             await self.uploadActor.remove(uploadID: id)
             self.notifyDelegates()
         }
     }
     
-    internal func findUploaderFor(videoFile url: URL) -> ChunkedFileUploader? {
-        return Dictionary<URL, ChunkedFileUploader>(
-            uniqueKeysWithValues: uploadersByID.mapValues { value in
-                (value.uploadInfo.videoFile, value)
-            }
-            .values
-        )[url]
+    internal func findChunkedFileUploader(
+        inputFileURL: URL
+    ) -> ChunkedFileUploader? {
+        findStartedUpload(
+            ofFile: inputFileURL
+        )?.fileWorker
     }
 
-    internal func registerUploader(_ fileWorker: ChunkedFileUploader, withId id: String) {
-        uploadersByID.updateValue(fileWorker, forKey: fileWorker.uploadInfo.id)
+    internal func registerUpload(_ upload: MuxUpload) {
+        guard let fileWorker = upload.fileWorker else {
+            // Only started uploads, aka uploads with a file
+            // worker can be registered.
+            // TODO: Should this throw?
```

Comment:
suggestion(non-blocking) good question. Maybe minimally have it log out a warning?

Reply:
8430449c4b510211cb0335b5b409447ef17d411e